user_management/doc/User_management.md

```diff
@@ -0,0 +1,136 @@
+# User Management
+
+--------------------------------------------------------------------------------
+
+## `signup`
+
+### Account creation
+
+will return a refresh token when successful
+
+<details>
+ <summary><code>POST</code><code><b>/user/signup/</b></code></summary>
+
+### Parameters
+
+#### Body
+all fields are mandatory
+- Username must be unique and between 1 and 20 characters long
+- Email must be unique and between 1 and 50 characters long
+- Password must be between 8 and 50 characters long and contain at least one uppercase letter, one digit and one special character
+> ``` javascript
+> {
+>     "username": "Aurel",
+>     "email": "alevra@student.42lyon.fr",
+>     "password": "Validpass42*"
+> }
+> ```
+
+#### Responses
+
+> | http code | content-type       | response                            |
+> |-----------|--------------------|-------------------------------------|
+> | `201`     | `application/json` | `{"refresh_token": "eyJhbGci.."}`   |
+> | `401`     | `application/json` | `{"errors": ["AAA", "BBB", "..."]}` |
+
+>errors can be combined
+
+> errors can be :
+> - Username already taken
+> - Username length {len(username)} > 20
```

Comment:
Missing 'Username length {len(username)} < {USERNAME_MIN_LENGTH}'

Reply:
Wouldn't be more convenient if the documentation itsel stated the min and max values for each fields ? ..
Rather than forcing each developer to look in the user_management/settings.py for each value

Reply:
Agreed on leaving the value inside the documentation